File: Solutions/2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.js
/**
 * @param {number[]} nums
 * @param {number} pivot
 * @return {number[]}
 */
 
var pivotArray = function(nums, pivot) {
    // Initialize three arrays to hold values less than, equal to, and greater than pivot
    let less = [];
    let equal = [];
    let greater = [];

    // Loop through nums and categorize each element
    for (let num of nums) {
        if (num < pivot) {
            less.push(num); // Elements less than pivot
        } else if (num === pivot) {
            equal.push(num); // Elements equal to pivot
        } else {
            greater.push(num); // Elements greater than pivot
        }
    }

    // Concatenate the arrays to form the result
    return [...less, ...equal, ...greater];
};

File: Solutions/2265-partition-array-according-to-given-pivot/partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        
        // Stores all elements smaller than pivot
        vector<int> less;
        
        // Stores all elements equal to pivot
        vector<int> equal;
        
        // Stores all elements greater than pivot
        vector<int> greater;
        
        // Traverse the original array
        for (int num : nums) {
            
            // If current element is smaller than pivot,
            // put it into the 'less' array
            if (num < pivot) {
                less.push_back(num);
            }
            
            // If current element is greater than pivot,
            // put it into the 'greater' array
            else if (num > pivot) {
                greater.push_back(num);
            }
            
            // Otherwise it must be equal to pivot
            else {
                equal.push_back(num);
            }
        }
        
        // Result array
        vector<int> ans;
        
        // First add all smaller elements
        for (int x : less) {
            ans.push_back(x);
        }
        
        // Then add all pivot elements
        for (int x : equal) {
            ans.push_back(x);
        }
        
        // Finally add all greater elements
        for (int x : greater) {
            ans.push_back(x);
        }
        
        // Return the rearranged array
        return ans;
    }
};